sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -35,6 +34,9 @@ namespace Azure.Messaging.EventHubs.Producer
     ///
     public class EventHubProducerClient : IAsyncDisposable
     {
+        /// <summary>The maximum number of attempts to send an event by a partition specific producer</summary>
+        internal const int MaximumNumberOfRetrials = 2;
```

Comment:
nit: Should use the value configured on the retry policy for this rather than having a second, special value?

Reply:
Good point, I added an extra property ```MaximumRetries``` to ```BasicRetryPolicy``` and ```EventHubsRetryPolicy```. I look forward to knowing your thinking about them.

Reply:
Well.... so, I'm the culprit here.   Yeah, sorry.   This was a bad thought on my part and I should have rotten fruit thrown at me for suggesting it.